gemelli/rpca.py

```diff
@@ -579,9 +579,12 @@ def frequency_filter(val, id_, md):
         return (np.sum(val > 0) / n_samples) > (min_feature_frequency / 100)
 
     # filter and import table for each filter above
-    table = table.filter(observation_filter, axis='observation')
-    table = table.filter(frequency_filter, axis='observation')
-    table = table.filter(sample_filter, axis='sample')
+    if min_feature_count is not None:
+        table = table.filter(observation_filter, axis='observation')
+    if min_feature_frequency is not None:
+        table = table.filter(frequency_filter, axis='observation')
+    if min_sample_count is not None:
+        table = table.filter(sample_filter, axis='sample')
```

Comment:
Suggest ensuring empty samples / features are removed just in case

```suggestion
        table = table.filter(sample_filter, axis='sample')
    table = table.remove_empty()
```

Reply:
Added! Thanks!